db/sql/0_msar.sql

```diff
@@ -2333,7 +2369,7 @@ BEGIN
     msar.get_relation_namespace_oid(tab_id),
     new_tab_name,
     extracted_col_defs,
```

Comment:
Now I see why I missed this bug when reviewing the column moving logic PR. I was assuming `extracted_col_defs` contained constraints too. 

`msar.process_col_def_jsonb`  actually handles `NOT NULL` constraint because it cannot be added as a table constraint. Handling constraints in two different places is pesky. We should add comments to the `msar.process_col_def_jsonb` function to clear up the confusion

Reply:
As I described above, the problem is actually that we're dealing with constraints in both ways in Python, and I wanted to unify it to table-only constraints. Irritatingly, `NOT NULL` isn't really implemented as a constraint at all in PostgreSQL. It's a column property (like its type or default). I.e., the nullability of a column is stored as a boolean in the `pg_attribute` table, not as a row in the `pg_constraint` table. So the fact that you can't store a null value in a given column is treated the same as the fact that you can't store a value of `'abc'` in a `numeric` column. However, SQLAlchemy, numerous online resources, and even the PostgreSQL docs often call `NOT NULL` a constraint, leading to this seeming inconsistency, and lots of confusion all around. Quite irritating.

I'll try to add some comments to that effect, though.